General Cleanup (#65)
CMakePP/CMakePPLang: docs/source/developer/unit_testing.rst

.. _unit-testing-cmakepplang-functions:

**********************************
Unit Testing CMakePPLang Functions
**********************************

This page contains guidelines, tips, and tricks which relate to unit testing
functionality in the CMakePPLang module. The content of this page is somewhat
exclusive to CMakePPLang and is in addition to project-wide guidelines.

Testing Signatures
==================

The CMake language is weakly typed, while CMakePPLang is strongly typed.
Because CMakePPLang is unit tested, users can be confident that CMakePPLang

> functions will correctly enforce type safety (and if they don't, that is a

Reviewer: Instead of functions "code"?  I think that makes it more clear this applies to say classes too.

> bug). Unfortunately, this means that all other functions intended to implement
> CMakePPLang functions need to have their type-safety checked manually. This
> section focuses on how to unit test the signatures of functions which are
> strongly typed, but are not CMakePPLang functions.

Reviewer: ```suggestion
bug). Unfortunately, this means that all functions intended to implement
CMakePPLang's type-safety mechanism need to have their type-safety checked manually. This section focuses on how to unit test such functions.
```
